deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/DepthwiseConvolution2D.java

```diff
@@ -0,0 +1,178 @@
+package org.deeplearning4j.nn.conf.layers;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.deeplearning4j.nn.api.Layer;
+import org.deeplearning4j.nn.api.ParamInitializer;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.layers.convolution.DepthwiseConvolution2DLayer;
+import org.deeplearning4j.nn.params.DepthwiseConvolutionParamInitializer;
+import org.deeplearning4j.optimize.api.TrainingListener;
+import org.deeplearning4j.util.ConvolutionUtils;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import java.util.*;
+
+/**
+ * 2D depth-wise convolution layer configuration.
+ * <p>
+ * Performs a channels-wise convolution, which
+ * operates on each of the input maps separately. A channel multiplier is used to
+ * specify the number of outputs per input map. This convolution
+ * is carried out with the specified kernel sizes, stride and padding values.
+ *
+ * @author Max Pumperla
+ */
+@Data
+@NoArgsConstructor
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+public class DepthwiseConvolution2D extends ConvolutionLayer {
+
+    int depthMultiplier;
+
+    protected DepthwiseConvolution2D(Builder builder) {
+        super(builder);
+        this.hasBias = builder.hasBias;
+        this.depthMultiplier = builder.depthMultiplier;
+        this.convolutionMode = builder.convolutionMode;
+        this.dilation = builder.dilation;
+        if (builder.kernelSize.length != 2)
+            throw new IllegalArgumentException("Kernel size of should be rows x columns (a 2d array)");
+        this.kernelSize = builder.kernelSize;
+        if (builder.stride.length != 2)
+            throw new IllegalArgumentException("Stride should include stride for rows and columns (a 2d array)");
+        this.stride = builder.stride;
+        if (builder.padding.length != 2)
+            throw new IllegalArgumentException("Padding should include padding for rows and columns (a 2d array)");
+        this.padding = builder.padding;
+        this.cudnnAlgoMode = builder.cudnnAlgoMode;
+        this.cudnnFwdAlgo = builder.cudnnFwdAlgo;
+        this.cudnnBwdFilterAlgo = builder.cudnnBwdFilterAlgo;
+        this.cudnnBwdDataAlgo = builder.cudnnBwdDataAlgo;
+        this.nOut = this.nIn * this.depthMultiplier;
+
+        initializeConstraints(builder);
+    }
+
+    public boolean hasBias() {
+        return hasBias;
+    }
+
+    @Override
+    public DepthwiseConvolution2D clone() {
+        DepthwiseConvolution2D clone = (DepthwiseConvolution2D) super.clone();
+        if (clone.kernelSize != null)
+            clone.kernelSize = clone.kernelSize.clone();
+        if (clone.stride != null)
+            clone.stride = clone.stride.clone();
+        if (clone.padding != null)
+            clone.padding = clone.padding.clone();
```

Comment:
Clone method is wrong. Kernel/stride/padding are already handled in superclass clone.
And it doesn't set depth multiplier.